commit 25b39703b2c3468f666f6175ff94e46526d402a7
Author: Xerxes Rånby <xerxes@gudinna.com>
Date:   Tue Dec 22 13:33:44 2015 +0100

    OpenJDK 8: Add JDK8u JDK-8061651 API. IcedTea PR2775.
    
    Implement minimal JVM functionality to tell JDK8u that the JDK-8061651 API is unsupported by JamVM
    
    JVM_GetResourceLookupCacheURLs
    JVM_GetResourceLookupCache
    JVM_KnownToNotExist
    
    Signed-off-by: Xerxes Rånby <xerxes@gudinna.com>

diff --git a/src/classlib/openjdk/jvm.c b/src/classlib/openjdk/jvm.c
index 971ac75..955621d 100644
--- jamvm.old/jamvm/src/classlib/openjdk/jvm.c
+++ jamvm/jamvm/src/classlib/openjdk/jvm.c
@@ -666,6 +666,35 @@ void JVM_SetClassSigners(JNIEnv *env, jclass cls, jobjectArray signers) {
 }
 
 
+/* JVM_GetResourceLookupCacheURLs
+   is part of the
+   JDK-8061651 JDK8u API
+*/
+
+jobjectArray JVM_GetResourceLookupCacheURLs(JNIEnv *env, jobject loader) {
+    return NULL; // tell OpenJDK 8 that the lookup cache API is unavailable
+}
+
+/* JVM_GetResourceLookupCache
+   is unused however it is part of the
+   JDK-8061651 JDK8u API
+*/
+
+jintArray JVM_GetResourceLookupCache(JNIEnv *env, jobject loader, const char *resource_name) {
+    UNIMPLEMENTED("JVM_GetResourceLookupCache");
+    return NULL; // tell OpenJDK 8 that the lookup cache is unavailable
+}
+
+/* JVM_KnownToNotExist
+   is unused however it is part of the
+   JDK-8061651 JDK8u API
+*/
+
+jboolean JVM_KnownToNotExist(JNIEnv *env, jobject loader, const char *classname) {
+    UNIMPLEMENTED("JVM_KnownToNotExist");
+    return JNI_FALSE; // tell OpenJDK 8 we don't know whether it exists or not
+}
+
 /* JVM_GetProtectionDomain */
 
 jobject JVM_GetProtectionDomain(JNIEnv *env, jclass cls) {
